src/Analyzer/Resolve/IdentifierResolver.cpp

```diff
@@ -187,10 +187,10 @@ std::shared_ptr<TableNode> IdentifierResolver::tryResolveTableIdentifier(const I
     if (!storage)
```

Comment:
Why didn’t you add a test like: 'N threads read from ClickHouse, while 1 thread modifies data in Spark and uploads data and metadata (strictly in this order)'? At least we would get a tsan check in CI…